### PR TITLE
Extract side effects introduced by System#getenv()

### PR DESCRIPTION
**Pull request by Weltraumschaf**
Extract the`system#getenv()` calls into own class as prerequsite for https://github.com/secureCodeBox/secureCodeBox/pull/2133.

### CODE REVIEW

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    return envVar.trim();
+  }
+
+
+  @Getter
+  enum EnvVarNames {
+    // TODO: Consider prefixing this name with DEFECTDOJO_.
+    IS_DEV("IS_DEV"),
```
Comment (Weltraumschaf) [inline hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:65]: @J12934 Should we prefix this also with `DEFECTDOJO_`?

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    // TODO: Consider prefixing this name with DEFECTDOJO_.
+    SCAN_NAME("SCAN_NAME"),
```
Comment (Weltraumschaf) [inline hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:67]: @J12934 Should we prefix this also with `DEFECTDOJO_`?

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    // TODO: Consider prefixing this name with DEFECTDOJO_.
+    NAMESPACE("NAMESPACE"),
```
Comment (Weltraumschaf) [inline hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:69]: @J12934 Should we prefix this also with `DEFECTDOJO_`?

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * This class provides an API to access configuration values from environment variables
+ */
+@Slf4j
+public final class EnvConfig {
+
+  public boolean isDev() {
```
Comment (Weltraumschaf) [inline hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:12]: @J12934 Do I understand right that this is kinda development mode to load a local k8s config from `~/.kube/config`?

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    return Boolean.parseBoolean(retrieveEnvVar(EnvVarNames.IS_DEV));
+  }
+
+  public String scanName() {
```
Comment (Weltraumschaf) [inline hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:16]: @J12934 What's the purpose of this config?

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    return retrieveEnvVar(EnvVarNames.SCAN_NAME);
+  }
+
+  public String namespace() {
```
Comment (Weltraumschaf) [inline hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:20]: @J12934 What's the purpose of this config?

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    return retrieveEnvVar(EnvVarNames.NAMESPACE);
+  }
+
+  public boolean lowPrivilegedMode() {
```
Comment (Weltraumschaf) [inline hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:24]: @J12934 What's the purpose of this config?

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java (hunk 4 above)
Comment (J12934) [reply hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:12]: yes

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java (hunk 7 above)
Comment (J12934) [reply hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:24]: https://www.securecodebox.io/docs/hooks/defectdojo#low-privileged-mode

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java (hunk 6 above)
Comment (J12934) [reply hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:20]: Its the kubernetes namespace the scan is running in.
Required for hooks to be able to update the status of the scan (e.g. if the changed the number of findings they should update the status of the scan (done automatically by the hook-sdk for all JS hooks (every hook but this one))
This is automatically set for every hook job

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java (hunk 3 above)
Comment (J12934) [reply hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:69]: No as this is consistent across all hooks and has nothing do do with defect dojo.
Might want to prefix it with SECURECODEBOX but i don't think it really required

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java (hunk 2 above)
Comment (J12934) [reply hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:67]: see: 
https://github.com/secureCodeBox/secureCodeBox/pull/2184#discussion_r1455906646

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java (hunk 1 above)
Comment (J12934) [reply hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:65]: This we can change to whatever we like as it's only used during development.

File: hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java (hunk 3 above)
Comment (Weltraumschaf) [reply hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:73]: Problem with such generic names is that name clashes with other tools is very likely and hard to debug. I suggest to deprecate `SCAN_NAME` and `NAME_SPACE` and prefix them with `SCB_`.

Comment (Weltraumschaf) [reply hooks/persistence-defectdojo/hook/src/main/java/io/securecodebox/persistence/config/EnvConfig.java:130]: https://github.com/secureCodeBox/secureCodeBox/issues/2185

**Review by J12934 — APPROVED** [review]

**Pull request merged**